assets/js/components/user-input/UserInputKeywords.js

```diff
@@ -57,9 +56,6 @@ export default function UserInputKeywords( { slug, max } ) {
 	// https://github.com/google/site-kit-wp/issues/2900#issuecomment-814843972.
 	const [ localValues, setLocalValues ] = useState( values );
 
-	// Need to make sure that dependencies list always has the same number of elements.
-	const dependencies = values.concat( Array( max ) ).slice( 0, max );
```

Comment:
I'm not following this. The dependencies array needs to have the dependencies so that the callback doesn't have stale values.